src/util/iterative_restoration.jl

```diff
@@ -101,37 +101,101 @@ end
+        ## This ensures that dispatches will be viable (especially storage) for the duration of the repair set
+
+        # save solve_time for accumulation
+        solve_time = restoration_solution["solve_time"]
+
+        # make a repaired devices list for processing
+        repaired_devices = Dict(comp_name=>String[] for comp_name in restoration_comps)
+        for (nw_id, list) in repairs
+            for (comp_name, comp_id) in list
+                push!(repaired_devices[comp_name],comp_id)
+            end
+        end
+        repair_count=sum(length(comp_ids) for (comp_name,comp_ids) in repaired_devices)
+
+        restoration_network = replicate_restoration_network(network, count=repair_count)
+
+        # set all repaired components inactive before final time period
+        for net_id in 0:repair_count-1
+            set_component_inactive!(restoration_network["nw"]["$(net_id)"], repaired_devices)
+        end
+        # clear damage indicator for final time period
+        for comp_name in restoration_comps
+            for (comp_id,comp) in restoration_network["nw"]["$(repair_count)"][comp_name]
+                comp["damaged"]=0
+            end
+        end
+
+        # run 'restoration' on this network to solve powerflow
+        restoration_solution = _run_rop_ir(restoration_network, model_constructor, optimizer, kwargs...)
+        restoration_solution["solve_time"]+=solve_time
+
+        # Was the network solved?
+        if restoration_solution["termination_status"]!= _MOI.OPTIMAL && restoration_solution["termination_status"]!= _MOI.LOCALLY_SOLVED
+            Memento.warn(_PM._LOGGER, "subnetwork i was not solved, returning current solution")
+            terminate_problem = true
+        else
+            terminate_problem = false
+        end
+
+        clean_solution!(restoration_solution)
+        _PM.update_data!(restoration_network, restoration_solution["solution"])
+        clean_status!(restoration_network)
+        _process_repair_status!(restoration_network)
+    end
 
+    # copy network metadata, remove network data. There should be a better way of doing this.
     subnet_solution_set = deepcopy(restoration_solution)
     subnet_solution_set["solution"]["nw"] = Dict{String,Any}()
+    # do not run restoration on network "0"
+    delete!(restoration_network["nw"],"0")
 
-    for(nw_id, network) in sort(Dict{Int,Any}([(parse(Int, k), v) for (k,v) in restoration_network["nw"]]))
-        if count_repairable_items(network) > 1
-            Memento.info(_PM._LOGGER, "sub_network $(nw_id) has $(count_damaged_items(network)) damaged items and $(count_repairable_items(network)) repairable items")
+    for(nw_id, net) in sort(Dict{Int,Any}([(parse(Int, k), v) for (k,v) in restoration_network["nw"]]))
+        net["time_elapsed"] = network["time_elapsed"] # is this the correct way to reset time-elapsed for each network?
+        if count_repairable_items(net) > 1 && ~terminate_problem && ~terminate_recursion
```

Comment:
Instead of `~` here I think you want to use `!`?

Reply:
I did not realize `~` was a bit wise not.